Update Rust tracker docs for 0.2.0

greg-el opened the pull request:
The PR adds docs for the release of [0.2.0 ](https://github.com/snowplow/snowplow-rust-tracker/pull/34)of the Rust tracker.

matus-tomlein commented on docs/collecting-data/collecting-from-own-applications/rust-tracker/getting-started/index.md:
````diff
@@ -31,7 +31,7 @@ use snowplow_tracker::Snowplow;
+let mut tracker = Snowplow::create_tracker("ns", "app_id", "https://...", None);
 ```
 
 There are additional optional arguments to configure the tracker. To learn more about configuring how events are sent, check out [this page](/docs/collecting-data/collecting-from-own-applications/rust-tracker/initialization-and-configuration/index.md).
````
Nit: would it make sense to also give a snippet how to deinitialize the tracker using the `close_emitter()` call here? I know it's mentioned on the following page but it's kind of buried and might be missed by some users.

greg-el replied:
Yeah, this seems like a good idea.

matus-tomlein reviewed (approved):
LGTM!

I don't know how I feel about duplicating the 0.1.0 docs under `previous-versions`. It seems a bit of an overkill since there are not that many breaking changes and we are still in the early releases. Maybe a short migration page would be enough to show what changed? It's quite annoying that when I search for something in the docs, it shows the results multiple times for each copy in previous versions...

stanch reviewed (changes requested):
You need to add a root `index.md` for `previous-versions` so that it displays as `Previous versions` — see other trackers for how it’s done. Also I have a slight preference for it to appear _before_ the API docs link.

<img width="260" alt="image" src="https://user-images.githubusercontent.com/2670454/208450960-016d99be-7586-4e86-a3fa-4ad9b75cd485.png">

greg-el commented:
I've removed the `previous-versions` anyway as Matus suggested, as the breaking changes are minimal. I've added a short migration guide "caution" on the index page for the tracker.

stanch commented on docs/collecting-data/collecting-from-own-applications/rust-tracker/index.md:
```diff
@@ -14,6 +14,14 @@ The Snowplow Rust Tracker allows you to add analytics to your Rust apps.
+
+If you are upgrading from 0.1.0 to 0.2.0, there are a couple of changes required to make to your code:
+
+- [`Tracker.track`](./getting-started/#tracking-events) is no longer an async function
```
```suggestion
- [`Tracker.track`](./getting-started/index.md#tracking-events) is no longer an async function
```
